### PR TITLE
Load resources with url and optional version

### PR DESCRIPTION
**Pull request by jingtang10**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #[issue number]

**Description**
Clear and concise code change description.

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: (Bug fix | Feature | Documentation | Testing | Code health | Builds | Releases | Other)

**Screenshots (if applicable)**

**Checklist**

- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://google.github.io/android-fhir/contrib/) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

File: knowledge/src/main/java/com/google/android/fhir/knowledge/KnowledgeManager.kt [hunk 1]
```diff
@@ -189,6 +191,65 @@ internal constructor(
...
+   *
+   * However, if a version is specified both as a parameter and as part of the URL, the two must
+   * match.
+   *
+   * @throws IllegalArgumentException if the url contains more than one pipe `|`
+   * @throws IllegalArgumentException if the version specified in the URL and the explicit version
+   *   do not match
+   */
+  suspend fun loadResources(
+    url: String,
```
Comment (MJ1998) [inline knowledge/src/main/java/com/google/android/fhir/knowledge/KnowledgeManager.kt:210]: I think we need to have resourceType as parameter. Take a look at this issue: https://github.com/google/android-fhir/issues/2490

**Review by MJ1998 — CHANGES_REQUESTED** [review]
Adding version looks find but why is resourceType being removed from apis here?

**Review by santosh-pingle — APPROVED** [review]
LGTM!

File: knowledge/src/main/java/com/google/android/fhir/knowledge/KnowledgeManager.kt (hunk 1 above)
Comment (jingtang10) [reply knowledge/src/main/java/com/google/android/fhir/knowledge/KnowledgeManager.kt:210]: canonical resources are universally identified using canonical urls: see https://hl7.org/fhir/resource.html#canonical and https://build.fhir.org/ig/HL7/crmi-ig/artifact-lifecycle.html#artifact-identity

we discussed this in great length with @brynrhodes (on android-fhir@google.com mailing list) and this is the conclusion we arrived at.

Comment (MJ1998) [reply knowledge/src/main/java/com/google/android/fhir/knowledge/KnowledgeManager.kt:210]: FYI @dubdabasoduba since you had raised the issue #2490

**Review by MJ1998 — APPROVED** [review]
LGTM

**Pull request merged**